base/uk.ac.stfc.isis.ibex.ui.scriptgenerator/src/uk/ac/stfc/isis/ibex/ui/scriptgenerator/views/ScriptGeneratorView.java

```diff
@@ -289,16 +292,17 @@ private void displayLoaded() {
 		        
 			    // Make buttons to move an action up and down the list
 		        Button btnMoveActionUp = createMoveRowButton(moveComposite, "move_up.png", "up");
-		        btnMoveActionUp.addListener(SWT.Selection, e -> {
-		        	scriptGeneratorViewModel.moveActionUp(table.getSelectionIndex());
-		        	table.setSelectionIndex(Math.max(0, table.getSelectionIndex() - 1));
+		        btnMoveActionUp.addListener(SWT.Selection, e ->	{
+		        	List<ScriptGeneratorAction> selection = table.selectedRows();
+		        	scriptGeneratorViewModel.moveActionUp(selection);
+			        table.viewer().setSelection((ISelection) selection);
 		        });
 		        
 		        Button btnMoveActionDown = createMoveRowButton(moveComposite, "move_down.png", "down");
 		        btnMoveActionDown.addListener(SWT.Selection, e -> {
-		        	scriptGeneratorViewModel.moveActionDown(table.getSelectionIndex());
-		            table.setSelectionIndex(Math.min(
-		            		table.getSelectionIndex() + 1, scriptGeneratorViewModel.getActions().size()));
+		        	List<ScriptGeneratorAction> selection = table.selectedRows();
+		        	scriptGeneratorViewModel.moveActionDown(selection);
+		        	table.viewer().setSelection((ISelection) selection);
```

Comment:
I'm getting the following exception on this line when selecting multiple rows and moving them up/down.

```
!ENTRY org.eclipse.ui 4 0 2020-06-19 14:34:15.174
!MESSAGE Unhandled event loop exception
!STACK 0
java.lang.ClassCastException: class java.util.ArrayList cannot be cast to class org.eclipse.jface.viewers.ISelection (java.util.ArrayList is in module java.base of loader 'bootstrap'; org.eclipse.jface.viewers.ISelection is in unnamed module of loader org.eclipse.osgi.internal.loader.EquinoxClassLoader @41a90fa8)
	at uk.ac.stfc.isis.ibex.ui.scriptgenerator.views.ScriptGeneratorView.lambda$9(ScriptGeneratorView.java:305)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:89)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4175)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1057)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:3988)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3587)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1160)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1049)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:155)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:660)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:559)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:154)
	at uk.ac.stfc.isis.ibex.scriptgenerator.product.Application.start(Application.java:47)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:203)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:137)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:107)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:401)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:255)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:657)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:594)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1465)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1438)
```